pulp_2to3_migration/app/plugin/rpm/repository.py

```diff
@@ -73,7 +73,8 @@ def migrate_to_pulp3(cls, pulp2distributor, repo_version):
             if pulp2_checksum_type:
                 checksum_types = {'metadata': pulp2_checksum_type,
                                   'package': pulp2_checksum_type}
-            publish(repo_version.pk, checksum_types=checksum_types)
+            sqlite = pulp2_config.get('generate_sqlite', False)
+            publish(repo_version.pk, checksum_types=checksum_types, sqlite_metadata=sqlite)
```

Comment:
Problem: Only the master branch of ```pulp_rpm``` has this feature, which means that if we provide this flag, it's going to fail (unexpected keyword argument) with any currently-available release of the plugin.  I'm not sure of the best way to address this.

* We could backport this feature to older releases of pulp_rpm so that it will work properly
* We could backport a change to the the signature of ```publish(...)``` to include ```**kwargs``` so that it will be ignored

Neither of those are great options.  I don't think we can just break compatibility with any pulp_rpm except for an unreleased version since Katello won't be using it.

Reply:
https://github.com/pulp/pulp_rpm/pull/1908/files#diff-e556922435d79157d5a5d7ebc182d611f4bcb3e441813d2266382ff6b15c9c8bL259

Reply:
I think backporting a feature is not really an option since it's against all our rules :)
kwargs sounds like an option to me, the only pain is releasing 3.7 and 3.8.z releases.

I'm not sure which version of pulp_rpm Katello is using. They can upgrade to any until it's compatible with pulpcore 3.7.
Other options which are  not good or hacky but I throw them here anyway:
 - declare compatibility with pulpcore >= 3.9 that will force pulp_rpm version to be at least 3.9, they one with the feature in it. (it might create many inconveniences with packaging in downstream and with our development. Since  we still need to have compatibility with 3.7 pulpcore, just pulp_rpm needs to be 3.9. Are there conditional dependencies in pip world? :) )
 - hack with try/except to support both signatures

Reply:
What Justin said:

> \<jsherrill> dalley: if pulp_rpm 3.9 is compatible with pulpcore 3.7, we can probably upgrade it in 3.7.  Assuming that 'upload' directory issue isn't a problem within pulp_rpm 3.7 (i assume thats a pulpcore issue?)
> \<jsherrill> err upgrade pulp_rpm to 3.9 i mean

Reply:
yeah, as expected, it doesn't help us though :)

Reply:
if understand the situation correctly, in case user has pulp_rpm 3.7, for example, no matter what we do, the sqlite will be generated, because pulp_rpm 3.7 code path does not any other option.
I would prefer a hack with the try/except to support both signatures and not fail. kwargs is less dirty approach however i don't know whether it is worth releasing 3.7.z and 3.8.z for the sake of this problem. I would image that in real world the user would anyway want to be on the latest release version before migrating all his data from pulp2 to pulp3.
And if the user is on older versions and does he not want the sqlite generated  he would not habe other choice then to upgrade first to pulp_rpm 3.9